src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5524,125 +5521,198 @@ private VariableDeclaratorSyntax ParseVariableDeclarator(
                 name = this.AddError(name, ErrorCode.ERR_MultiTypeInDeclaration);
             }
 
-            switch (this.CurrentToken.Kind)
+            return this.CurrentToken.Kind switch
             {
-                case SyntaxKind.EqualsToken:
-                    if (isFixed)
-                    {
-                        goto default;
-                    }
+                SyntaxKind.EqualsToken when !isFixed => parseNonFixedVariableDeclaratorWithEqualsToken(name, argumentList: null, out localFunction),
```

Comment:
prevents a confusing jump inside parseNonFixedVariableDeclaratorWithEqualsToken to parseVariableDeclaratorDefault.  This simplifies things as we don't have to wonder how an argumentList might flow through all codepaths to these helpers.